### PR TITLE
Added migro

### PR DESCRIPTION
**Pull request by aisrael**
I wrote migro as I wanted a better SQL migration tool than e.g. Liquibase or Flyway. While there are existing database migration tools written in Crystal already, they're limited to migrations written in SQL.

With migro, you can specify migrations in YAML or SQL. In the future, I hope to support automatic rollback for migrations written in YAML (ala ActiveRecord Migrations).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -187,6 +187,7 @@ Contributions are welcome. Please take a quick look at the [contribution guideli
  * [crecto-admin](https://github.com/Crecto/crecto-admin) - Admin dashboard for Crecto and your database
  * [micrate](https://github.com/juanedi/micrate) - Database migration tool
  * [migrate](https://github.com/vladfaust/migrate.cr) - A simpler database migration tool with transactions
+ * [migro](https://github.com/aisrael/migro) - A database migration tool written in Crystal
```
Comment (veelenga) [inline README.md]: Please remove "in Crystal" part as it is implied here. Would be nice if you could write about `YML` directly in this title, so the user will see the difference between these tools here, on awesome-crystal page.

**Comment by RX14** [comment]
Looks like cql opens a new connection to the database for every query. That'll get you horrible performance. A single DB instance should be kept around and it does the pooling for you:

https://github.com/aisrael/cql/blob/dcad61b45838c7f78b53a10bd6d41d6fc945b591/src/cql/database.cr#L40

**Comment by RX14** [comment]
Also this is a SQL injection: https://github.com/aisrael/cql/blob/dcad61b45838c7f78b53a10bd6d41d6fc945b591/src/cql/postgresql.cr#L15

even for `table_exists?` you should be able to use a `$1` placeholder to get proper escaping.

**Comment by aisrael** [comment]
@RX14 Thanks! I've taken your suggestion about SQL injection and parameterised `table_exists?` as well as added some table name checking in `CreateTable` to help avoid it.

**Comment by RX14** [comment]
@aisrael DB pooling is SUPER simple: just add
```cr
@db = DB.open(@url)
```
[here](https://github.com/aisrael/cql/blob/dcad61b45838c7f78b53a10bd6d41d6fc945b591/src/cql/database.cr#L9)

then remove `with_db` and `s/db/@db` and you're done.

Pooling is automatic and built in to the DB drivers.

**Pull request merged**